Make streaming 3d, 6d & skeleton optional.

EmilRosenquist opened the pull request:
You can now choose which components you want to stream. (Timecode is always streamed)
---
Added the following options to the create live link source menu.

![image](https://user-images.githubusercontent.com/37864286/99225823-f8808400-27e8-11eb-8994-615ec481a8fe.png)

EmilRosenquist committed:
Make streaming 3d, 6d & skeleton optional.

helmesjo commented:
Add spaces to GUI strings while we are at it.

mlveggo commented:
Lets call 3d -> markers and 6d -> rigid bodies

mlveggo committed:
Switched 3d->markers, 6d->rigid bodies

helmesjo commented:
Default `3d`/`markers` to off. Or default all except `skeleton` to off?

helmesjo commented on Qualisys/QTMConnectLiveLink/Source/QTMConnectLiveLink/Private/QTMConnectLiveLinkSource.cpp:
```diff
@@ -36,13 +36,37 @@ QTMConnectLiveLinkSettings QTMConnectLiveLinkSettings::FromString(const FString&
 {
     FString settingsString = FString::Printf(TEXT("IpAddress=\"%s\""), *IpAddress);
     settingsString.Append(FString::Printf(TEXT("AutoDiscover=\"%d\""), AutoDiscover ? TEXT("true") : TEXT("false")));
+    settingsString.Append(FString::Printf(TEXT("Stream3d=\"%d\""), Stream3d ? TEXT("true") : TEXT("false")));
+    settingsString.Append(FString::Printf(TEXT("Stream6d=\"%d\""), Stream6d ? TEXT("true") : TEXT("false")));
```
Change GUI labels to `Stream Markers` and `Stream Rigid Bodies`.

mlveggo replied:
That is not a gui label. The settings are sent as a string between the livelink editor and livelink plugin. Don't need to change internal naming.

mlveggo commented:
6dof and skeleton on, markers can be off. Acting usually involves 6dof props.

helmesjo commented on Qualisys/QTMConnectLiveLink/Source/QTMConnectLiveLink/Private/QTMConnectLiveLinkSource.cpp:
```diff
@@ -36,13 +36,37 @@ QTMConnectLiveLinkSettings QTMConnectLiveLinkSettings::FromString(const FString&
     settingsString.Append(FString::Printf(TEXT("AutoDiscover=\"%d\""), AutoDiscover ? TEXT("true") : TEXT("false")));
+    settingsString.Append(FString::Printf(TEXT("Stream3d=\"%d\""), Stream3d ? TEXT("true") : TEXT("false")));
+    settingsString.Append(FString::Printf(TEXT("Stream6d=\"%d\""), Stream6d ? TEXT("true") : TEXT("false")));
+    settingsString.Append(FString::Printf(TEXT("StreamSkeleton=\"%d\""), StreamSkeleton ? TEXT("true") : TEXT("false")));
```
Change GUI label to `Stream Skeleton`.

mlveggo replied:
This is not a gui string

helmesjo commented on Qualisys/QTMConnectLiveLink/Source/QTMConnectLiveLink/Private/QTMConnectLiveLinkSource.cpp:
```diff
@@ -36,13 +36,37 @@ QTMConnectLiveLinkSettings QTMConnectLiveLinkSettings::FromString(const FString&
         if (autoDiscover == "true")
             settings.AutoDiscover = true;
     }
+    settings.Stream3d = true;
+    FString stream3d;
+    if (!FParse::Value(*settingsString, TEXT("Stream3d="), stream3d))
+    {
+        if (stream3d == "false")
+            settings.Stream3d = false;
+    }
+    settings.Stream6d = true;
+    FString stream6d;
+    if (!FParse::Value(*settingsString, TEXT("Stream6d="), stream6d))
+    {
+        if (stream6d == "false")
+            settings.Stream6d = false;
+    }
+    settings.StreamSkeleton = true;
+    FString streamSkeleton;
+    if (!FParse::Value(*settingsString, TEXT("StreamSkeleton="), streamSkeleton))
+    {
+        if (streamSkeleton == "false")
+            settings.StreamSkeleton = false;
+    }
```
Default all but `StreamSkeleton` to `false`.

mlveggo replied:
Rather stream3d to false... so we don't default stream markers.

helmesjo commented on Qualisys/QTMConnectLiveLink/Source/QTMConnectLiveLink/Public/QTMConnectLiveLinkBlueprint.h:
```diff
@@ -12,6 +12,6 @@ class QTMCONNECTLIVELINK_API UQTMConnectLiveLinkBlueprint : public UBlueprintFun
 
-    UFUNCTION(BlueprintCallable, Category = "Qualisys", meta = (DisplayName = "Create QTM Connect LiveLink Source", IpAddress = "127.0.0.1", AutoDiscover = "true"))
-    static void CreateQTMConnectLiveLinkSource(FString IpAddress, bool AutoDiscover, FLiveLinkSourceHandle& SourceHandle);
+    UFUNCTION(BlueprintCallable, Category = "Qualisys", meta = (DisplayName = "Create QTM Connect LiveLink Source", IpAddress = "127.0.0.1", AutoDiscover = "true", Stream3d = "true", Stream6d = "true", StreamSkeleton = "true"))
+    static void CreateQTMConnectLiveLinkSource(FString IpAddress, bool AutoDiscover, bool Stream3d, bool Stream6d, bool StreamSkeleton, FLiveLinkSourceHandle& SourceHandle);
```
Default all but `StreamSkeleton` to `false`.

helmesjo commented on Qualisys/QTMConnectLiveLink/Source/QTMConnectLiveLink/Public/QTMConnectLiveLinkSource.h:
```diff
@@ -18,7 +18,7 @@ class CRTProtocol;
 {
 public:
-	QTMConnectLiveLinkSettings() : IpAddress("127.0.0.1"), AutoDiscover(false)
+	QTMConnectLiveLinkSettings() : IpAddress("127.0.0.1"), AutoDiscover(false), Stream3d(true), Stream6d(true), StreamSkeleton(true)
```
Default all but `StreamSkeleton` to `false`.

mlveggo replied:
Rather stream3d to false

helmesjo replied:
Why was this marked as resolved? stream3d is still true. Or is the default value set elsewhere?

EmilRosenquist committed:
Set stream markers as default off and add space.

EmilRosenquist commented:
> Add spaces to GUI strings while we are at it.

> Lets call 3d -> markers and 6d -> rigid bodies

Fixed in commits [d025dac3](https://github.com/qualisys/QTM-Connect-For-Unreal/pull/9/commits/d025dac38ce9cfc980715482d97ea123330137db) and [638a295](https://github.com/qualisys/QTM-Connect-For-Unreal/pull/9/commits/638a2958ccbe9e7934a623215607b4a978948611).

> Default `3d`/`markers` to off. Or default all except `skeleton` to off?

> 6dof and skeleton on, markers can be off. Acting usually involves 6dof props.

Fixed in commit [638a295](https://github.com/qualisys/QTM-Connect-For-Unreal/pull/9/commits/638a2958ccbe9e7934a623215607b4a978948611).
